src/Chainweb/Mempool/RestAPI/Server.hs

```diff
@@ -28,24 +27,38 @@ import Chainweb.Mempool.Mempool
 import Chainweb.Mempool.RestAPI
 import Chainweb.RestAPI.Orphans ()
 import Chainweb.RestAPI.Utils
+import Chainweb.Time (getCurrentTimeIntegral)
 import Chainweb.Utils
 import Chainweb.Version
 
 ------------------------------------------------------------------------------
-insertHandler :: Show t => MempoolBackend t -> [T.Text] -> Handler NoContent
-insertHandler mempool txsT = handleErrs (NoContent <$ begin)
+insertHandler
+    :: forall t
+    .  ChainwebVersion
+    -> MempoolBackend t
+    -> [T.Text]
+    -> Handler NoContent
+insertHandler v mempool txsT = handleErrs (NoContent <$ begin)
   where
     txcfg = mempoolTxConfig mempool
+
+    decode :: T.Text -> Either String t
     decode = codecDecode (txCodec txcfg) . T.encodeUtf8
 
+    go :: T.Text -> Handler t
     go h = case decode h of
         Left e -> throwM . DecodeException $ T.pack e
         Right t -> return t
 
+    begin :: Handler ()
     begin = do
-        txs <- mapM go txsT
-        let txV = V.fromList txs
-        liftIO $ mempoolInsert mempool CheckedInsert txV
+        now <- liftIO getCurrentTimeIntegral
+        case txActivationDate v of
+            Just start | start < now -> pure ()
+            _ -> do
+                txs <- mapM go txsT
+                let txV = V.fromList txs
+                liftIO $ mempoolInsert mempool CheckedInsert txV
```

Comment:
This needs a comment, that allows to identify the code via `grep` and remove it in the first release after Dec 6.